yb-voyager/src/query/queryissue/detectors_ddl.go

```diff
@@ -163,6 +163,25 @@ func (d *TableIssueDetector) DetectIssues(obj queryparser.DDLObject) ([]QueryIss
 						))
 				}
 			}
+
+			if c.ConstraintType == queryparser.PRIMARY_CONSTR_TYPE {
+				//Report PRIMARY KEY (createdat timestamp) as hotspot issue
+				if len(c.Columns) > 0 {
+					col := c.Columns[0] // checking the first column only.
+					columnWithHotspotTypes, tableHasHotspotTypes := d.columnsWithHotspotRangeIndexesDatatypes[table.GetObjectName()]
+					if tableHasHotspotTypes {
+						//If first column is hotspot type then only report hotspot issue
+						hotspotTypeName, isHotspotType := columnWithHotspotTypes[col]
+						if isHotspotType {
+							hotspotIssues, err := reportHotspotsOnTimestampTypes(hotspotTypeName, table.GetObjectType(), table.GetObjectName(), col, false)
+							if err != nil {
+								return nil, err
+							}
+							issues = append(issues, hotspotIssues...)
+						}
+					}
+				}
+			}
```

Comment:
I'd recommend a slight restructure in-line with early-return pattern. It looks too nested right now. 
```
if PK : 
    if columns < 0 : 
        log-warn; continue;
    columnWithHotspotTypes, tableHasHotspotTypes = .... 
    if !tableHasHotspotTypes: 
        continue;
    ... (check if column is hotspot type; create issue if required)
```